pkg/engine/template_generator.go

```diff
@@ -640,6 +640,43 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		"GetB64sshdConfig": func() string {
 			return getBase64CustomScript(sshdConfig)
 		},
+		"GetSshPublicKeys": func() string {
+			// This generates the publicKeys array described at https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/createorupdate#sshconfiguration
+			// "ssh": {
+			//     "publicKeys": [
+			//       {
+			//         "keyData": "[parameters('sshRSAPublicKey')]",
+			//         "path": "[variables('sshKeyPath')]"
+			//       }
+			//     ]
+			//   }
+			str := ""
```

Comment:
@tariq1890 do you have any thoughts on a more maintainable implementation? Will a `fmt.Sprintf(...)` w/ three backticks type usage pattern make this less hairy?

Reply:
I think we can use a struct and marshal that into a JSON array.

Reply:
ok, switching to the struct that's vendored from the Azure SDK. It actually makes the code longer but if the API changes it could prevent breakage in the future.

Reply:
yay teamwork

Reply:
@tariq1890 - can you review now? Is that what you had in mind?